solr/modules/s3-repository/build.gradle

```diff
@@ -41,21 +41,25 @@ dependencies {
   runtimeOnly(group: 'com.fasterxml.woodstox', name: 'woodstox-core')
   runtimeOnly(group: 'org.codehaus.woodstox', name: 'stax2-api')
 
-  testImplementation project(':solr:test-framework')
-  testImplementation 'org.apache.lucene:lucene-test-framework'
-  testImplementation 'com.carrotsearch.randomizedtesting:randomizedtesting-runner'
   testImplementation 'junit:junit'
+  testImplementation 'com.carrotsearch.randomizedtesting:randomizedtesting-runner'
+  testImplementation 'org.apache.lucene:lucene-test-framework'
+  testImplementation project(':solr:test-framework')
+
+  // overrides spring-boot pulled by com.adobe.testing:s3mock-junit4 to allow using a version
+  // compatible with recent Jetty
+  testImplementation 'org.springframework.boot:spring-boot'
 
   testImplementation('com.adobe.testing:s3mock-junit4') {
     // Don't pull in separate versions of these libs, just use what Solr already has
     exclude group: 'org.apache.logging.log4j', module: 'log4j-to-slf4j'
     exclude group: 'ch.qos.logback', module: 'logback-classic'
     exclude group: 'org.eclipse.jetty', module: '*'
     exclude group: 'io.netty', module: '*'
-    exclude group: 'commons-io', module: 'commons-io'
+    exclude group: 'software.amazon.awssdk', module: 'netty-nio-client'
     exclude group: 'commons-codec', module: 'commons-codec'
+    exclude group: 'commons-io', module: 'commons-io'
     exclude group: 'org.apache.commons', module: 'commons-lang3'
-    exclude group: 'software.amazon.awssdk', module: 'netty-nio-client'
```

Comment:
The rearranging here looks weird

Reply:
Not really sure why that is there, probably conflict merging over time and changes.